domain/blockdevice/state/types.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package state
+
+import (
+	"github.com/juju/errors"
+
+	"github.com/juju/juju/domain/blockdevice"
+)
+
+// These structs represent the persistent block device entity schema in the database.
+
+type BlockDevice struct {
+	ID string `db:"uuid"`
+
+	DeviceName     string `db:"name"`
+	Label          string `db:"label"`
+	DeviceUUID     string `db:"device_uuid"`
+	HardwareId     string `db:"hardware_id"`
+	WWN            string `db:"wwn"`
+	BusAddress     string `db:"bus_address"`
+	Size           uint64 `db:"size"`
```

Comment:
Based on the comments below that you have added (thank you). I think we should consider a DDL change to include the unit of size in the name so it is explicit the value it is carrying.

My initial assumption was that this was number of bytes but it isn't. As it is not a base SI unit I think we should be explicit. But just my two cents.

Reply:
agree